HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,343 @@
+                                rotationMatrixFromRollPitchYaw,
+                                representations.rotations[i].convention);
+    }
+
+    // Rotation Vector to Axis-angle
+    std::cout << "\nConverting from rotation vector to axis-angle" << std::endl;
+    const Eigen::AngleAxisd axisAngle =
+        Eigen::AngleAxisd(representations.rotationVector.norm(), representations.rotationVector.normalized());
+    std::cout << "\nAxis-angle:\n" << reorderAxisAngle(axisAngle) << std::endl;
+
+    // Axis-Angle to Quaternion
+    std::cout << "\nConverting axis-angle to quaternion" << std::endl;
+    const Eigen::Quaterniond quaternion = Eigen::Quaterniond(axisAngle);
+    std::cout << "\nQuaternion:\n" << quaternion.coeffs() << std::endl;
+
+    // Quaternion to Rotation Matrix
+    std::cout << "\nConverting quaternion to rotation matrix" << std::endl;
+    const Eigen::Matrix3d rotationMatrixFromQuaternion = quaternion.toRotationMatrix();
+    std::cout << "\nRotation Matrix:\n" << rotationMatrixFromQuaternion << std::endl;
+    fileStorageOut.write("RotationMatrixFromQuaternion", eigenToCv(rotationMatrixFromQuaternion));
+
+    // Calculate error in transformation (Rotation Matrix --> Quaternion --> Rotation Matrix)
+    std::cout << "\nVerifying Rotation Matrix from Quaternion" << std::endl;
+    compareRotation(rotationMatrix, rotationMatrixFromQuaternion);
+
+    fileStorageOut.release();
+}
+
+Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+{
+    Eigen::Vector4d reorderedAxisAngle;
+
+    for(int i = 0; i < 3; i++)
+        reorderedAxisAngle[i] = axisAngle.axis()[i];
+    reorderedAxisAngle[3] = axisAngle.angle();
+
+    return reorderedAxisAngle;
+}
+
+/** Converts Rotation Matrix to RollPitchYaw.
+    * The convention we use here is that Roll is a rotation about x-axis,
+    * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+    * @param rotationMatrix Rotation Matrix to convert to RPY
+    * @param rotation Enum to indicate which rotation convention to follow for this conversion
+    * @return An array of angles, in radians
+    */
+Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d &rotationMatrix,
+                          const RotationConvention &rotation = RotationConvention::ZYX_Intrinsic)
+{
+    Eigen::Array3d rollPitchYaw;
+    switch(rotation)
+    {
+        case RotationConvention::XYZ_Intrinsic: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+        case RotationConvention::XYZ_Extrinsic: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+        case RotationConvention::ZYX_Intrinsic: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+        case RotationConvention::ZYX_Extrinsic: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+        default: throw std::invalid_argument("Invalid rotation");
+    }
+    // The returned array should be ordered by Ox, Oy and Oz, where Oa is the rotation about axis a.
+    switch(rotation)
```

Comment:
Would it be simpler if you moved the reversal into the above `switch`? Then you don't need two switches, and all the things that happen to the data in a particular case is co-located in the source. Also you avoid the warning you currently get about not enumerating all enum values in the second switch.

If you do this, you can also return directly from inside the switch, and avoid having the temporary `rollPitchYaw` variable which you first default initialize and then overwrite in the switch.

I didn't test this, but maybe the entire function could be reduced to:
```c++
    switch(rotation)
    {
        case RotationConvention::XYZ_Intrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
        case RotationConvention::XYZ_Extrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
        case RotationConvention::ZYX_Intrinsic: return rotationMatrix.eulerAngles(2, 1, 0).reverse();
        case RotationConvention::ZYX_Extrinsic: return rotationMatrix.eulerAngles(0, 1, 2);
        default: throw std::invalid_argument("Invalid rotation");
    }
```